### PR TITLE
Make completion window move to top when cursor is below half

### PR DESCRIPTION
**Pull request by wojciechkepka**
Closes: #307

image

### CODE REVIEW

**Comment by archseer** [comment]
Neat! Did you test that it works correctly with splits? A vertical split for example

**Comment by wojciechkepka** [comment]
Vertical does seem to work, but with horizontal it is still displayed on the bottom. I will push a fix soon

**Comment by wojciechkepka** [comment]
Now it correctly displays even with splits.

File: helix-view/src/tree.rs [hunk 1]
```diff
@@ -434,6 +434,10 @@ impl Tree {
             self.focus = key;
         }
     }
+
+    pub fn full_area(&self) -> Rect {
```
Comment (archseer) [inline helix-view/src/tree.rs:438]: Call it either area or viewport to match terminology elsewhere

Comment (wojciechkepka) [reply helix-view/src/tree.rs]: Right, done.

**Comment by wojciechkepka** [comment]
I'm pretty sure it works as expected now, done tests with multiple split confiurations and it is correctly either displayed on top or bottom so that the cursor is always visible.

**Review by archseer — APPROVED** [review]

**Pull request merged**